source/lac/sparse_direct.cc

```diff
@@ -1359,14 +1370,47 @@ SparseDirectMUMPS::vmult(VectorType &dst, const VectorType &src) const
       // Copy solution into the given vector
       // For MUMPS with centralized solution (icntl[20]=0), the solution is only
       // on the root process (0) and needs to be distributed to all processes
-      if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+
+      // Get locally owned range for this process
+      const IndexSet &locally_owned = dst.locally_owned_elements();
+      const size_type local_size    = locally_owned.n_elements();
+
+      std::vector<double> local_values(local_size);
```

Comment:
Just move this line down to where you first assign anything to it. You can then also make it `const` and you don't need to provide the size here -- `scatter()` will create an object of the right size for you.